bokehjs/src/lib/models/glyphs/circle.ts

```diff
@@ -144,7 +154,15 @@ export class CircleView extends XYGlyphView {
     const candidates = this.index.indices({x0, x1, y0, y1})
 
     const indices: number[] = []
-    if (this._radius != null && this.model.properties.radius.units == "data") {
+    if (this._hit_radius != null) {
+      for (const i of candidates) {
+        const r2 = this._hit_radius[i]**2
+        const dist = (this.sx[i] - sx)**2 + (this.sy[i] - sy)**2
+        if (dist <= r2) {
+          indices.push(i)
+        }
+      }
+    } else if (this._radius != null && this.model.properties.radius.units == "data") {
```

Comment:
I think all the codepaths are over-complicated (partially by the `_hit_radius` and `max_hit_radius` things, which again should go away. I think there is only one code path, which is essentially the original code path. except instead of just comparing the point to `sradius[i]**2` it should compare to something like `(sradius[i] + this.model.hit_radius)**2`